test/e2e-common.sh

```diff
@@ -101,3 +151,63 @@ function dump_extra_cluster_state() {
     done
   done
 }
+
+function install_istio {
+  if [[ -z "${ISTIO_VERSION}" ]]; then
+    readonly ISTIO_VERSION="latest"
+  fi
+  echo ">> Installing Istio: ${ISTIO_VERSION}"
+
+  local istio_base="./third_party/istio-${ISTIO_VERSION}"
+  INSTALL_ISTIO_CRD_YAML="${istio_base}/istio-crds.yaml"
+  INSTALL_ISTIO_YAML="${istio_base}/istio-minimal.yaml"
+
+  echo "Istio CRD YAML: ${INSTALL_ISTIO_CRD_YAML}"
+  echo "Istio YAML: ${INSTALL_ISTIO_YAML}"
+
+  echo ">> Bringing up Istio"
+  echo ">> Running Istio CRD installer"
+  kubectl apply -f "${INSTALL_ISTIO_CRD_YAML}" || return 1
+  wait_until_batch_job_complete istio-system || return 1
+  UNINSTALL_LIST+=( "${INSTALL_ISTIO_CRD_YAML}" )
+
+  echo ">> Running Istio"
+  kubectl apply -f "${INSTALL_ISTIO_YAML}" || return 1
+  UNINSTALL_LIST+=( "${INSTALL_ISTIO_YAML}" )
+}
+
+# Installs Knative Serving in the current cluster, and waits for it to be ready.
+function install_knative_serving {
+  install_istio
+  echo ">> Installing Knative serving"
+  readonly SERVING_YAML="https://github.com/knative/serving/releases/download/${LATEST_RELEASE_VERSION}/serving.yaml"
```

Comment:
I don't think we should do that. I think here we are testing eventing. We should test it against stable environment, that count serving as well. In my mind we shouldn't install lastest serving here at any point. Btw. Only eventing is being upgraded and downgraded, serving is there all the time.

Reply:
it would be good to re-use existing functionality where possible.  I'd suggest either using the test-infra shell scripts as already suggested, or adding this kind of function there so all the knative repos could make use of it (in a different PR).

Reply:
Okay. I've done just that. I think I will update test-infra functions with separate PR.